generators/java/index.js

```diff
@@ -0,0 +1,202 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const util = require('util')
+const {default: chalk} = require('chalk')
+const yosay = require('yosay')
+const inquirer = require('inquirer')
+
+const Generator = require('yeoman-generator')
+
+const readdir = util.promisify(fs.readdir)
+
+module.exports = class extends Generator {
+  constructor(args, opts) {
+    super(args, opts)
+
+    this.option('skip-welcome', {
+      desc: 'Skip the welcome message.',
+      type: Boolean,
+      default: false,
+      hide: true
+    })
+  }
+
+  async prompting() {
+    if (!this.options.skipWelcome) {
+      this.log(yosay(`Welcome to the ${chalk.hex('#15bfff').bold.underline('SmartThings')} generator!`))
+    }
+
+    this.appConfig = {}
+
+    async function required(input) {
+      return input ? true : 'This field is required.'
+    }
+
+    Object.assign(this.appConfig, await this.prompt([{
+      type: 'input',
+      name: 'applicationName',
+      validate: required,
+      message: this._fmtMsg('What is the name your application in plain English (for documentation)?')
```

Comment:
Yeah, not sure why I put "English". What I was trying to get at was that this the regular name with spaces and such. Maybe just "What is the name your application for documentation?"?

Reply:
I would follow what Dev Workspace does: 

> This name will be used in the SmartThings app. Please provide a name that is friendly to the end user.

This can be condensed down to `Please provide a name that is friendly to the end user: ___`

Reply:
done 👍